src/Microsoft.ServiceFabric.Actors/Runtime/ActorStateProviderHelper.cs

```diff
@@ -435,18 +432,24 @@ internal static IActorStateProvider CreateDefaultStateProvider(ActorTypeInformat
                 return Task.FromResult(actorQueryResult);
             }
 
-            // Skip the previous returned entries
-            while (currentActorCount < previousActorCount)
+            // Find continuation point for enumeration

```

Comment:
You will need to make ReliableCollectionsActorStateProvider as well to support similar enumeration semantics for RC.

Reply:
Added the changes in ReliableCollectionsActorStateProvider as well